Jantimizer/QueryTestSuite/TestRunners/TestRunner.cs

```diff
@@ -33,45 +34,106 @@ public async Task<List<AnalysisResult>> Run(bool runParallel = false)
         }
 
-        private async Task<List<AnalysisResult>> RunQueriesSerial()
+        private async Task<List<TestCase>> RunQueriesSerial()
         {
-            var queryAnalysisResults = new List<AnalysisResult>();
+            var testCases = new List<TestCase>();
             foreach (FileInfo queryFile in CaseFiles)
             {
                 Console.WriteLine($"Running {queryFile}");
-                queryAnalysisResults.Add(DatabaseModel.Parser.ParsePlan(await DatabaseModel.Connector.AnalyseQuery(queryFile)));
+                TestCase testCase = await GenerateAndRunTestCase(queryFile);
+                testCases.Add(testCase);
+            }
+            return testCases;
+        }
+
+        private Task<TestCase> GenerateAndRunTestCase(FileInfo queryFile)
+        {
+            string testName = queryFile.Name;
+            string testCategory;
+            try
+            {
+                testCategory = queryFile.Directory.Parent.Name;
+            } catch (Exception ex)
+            {
+                testCategory = "N/A";
+                Console.WriteLine($"Could not get test caregory - {queryFile}");
             }
-            return queryAnalysisResults;
+            
+            TestCase testCase = new TestCase(testName, testCategory);
+            return testCase.Run(DatabaseModel.Parser.ParsePlan(DatabaseModel.Connector.AnalyseQuery(queryFile)));
         }
 
+        private void WriteResultToConsole()
+        {
+            foreach(var testCase in Results)
+                Console.WriteLine($"{DatabaseModel.Name} | {testCase.Category} | {testCase.Name} | Database predicted cardinality: [{(testCase.AnalysisResult.EstimatedCardinality)}], actual: [{testCase.AnalysisResult.ActualCardinality}]");
+        }
 
-        public async Task Cleanup()
+        private void SaveResult()
         {
-            await DatabaseModel.Connector.CallQuery(CleanupFile);
+            string directory = $"Results\\{TimeStamp.ToString("yyyy/MM/dd/HH.mm.ss")}";
+            string resultFileName = directory + "\\result.csv";
+            Directory.CreateDirectory(directory);
+            FileStream stream;
+            CsvConfiguration config;
+            if (File.Exists(resultFileName))
+            {
+                stream = File.Open(resultFileName, FileMode.Append);
+                config = new CsvConfiguration(CultureInfo.InvariantCulture)
+                {
+                    // Don't write the header again.
+                    HasHeaderRecord = false,
+                };
+
+            } else
+            {
+                stream = File.Open(resultFileName, FileMode.Create);
+                config = new CsvConfiguration(CultureInfo.InvariantCulture);
+            }
+            using (var writer = new StreamWriter(stream))
+            using (var csv = new CsvWriter(writer, config))
+            {
+                csv.WriteRecords(Results);
+                writer.Flush();
+            }
+
         }
```

Comment:
Would it perhaps be more fitting to have the CSV logging in a seperate class instead? Since it dont really have much to do with this class in general